// @uiw/react-prismjs/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ReactPrismjs;

var _regeneratorRuntime2 = _interopRequireDefault(require("@babel/runtime/helpers/regeneratorRuntime"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _prismjs = _interopRequireDefault(require("prismjs"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "language", "source", "children"];

function ReactPrismjs() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

  var codeRef = /*#__PURE__*/_react.default.createRef();

  var prefixCls = props.prefixCls,
      className = props.className,
      language = props.language,
      source = props.source,
      children = props.children,
      others = (0, _objectWithoutProperties2.default)(props, _excluded);
  var langCls = language ? "language-".concat(language) : '';

  function highlight() {
    return _highlight.apply(this, arguments);
  }

  function _highlight() {
    _highlight = (0, _asyncToGenerator2.default)( /*#__PURE__*/(0, _regeneratorRuntime2.default)().mark(function _callee() {
      return (0, _regeneratorRuntime2.default)().wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              if (codeRef.current) {
                _prismjs.default.highlightElement(codeRef.current);
              }

            case 1:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return _highlight.apply(this, arguments);
  }

  (0, _react.useEffect)(function () {
    highlight(); // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [language, source]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("pre", (0, _objectSpread2.default)((0, _objectSpread2.default)({
    className: "".concat(prefixCls, " ").concat(className || '', " ").concat(langCls)
  }, others), {}, {
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("code", {
      className: langCls,
      ref: codeRef,
      children: source || children
    })
  }));
}

ReactPrismjs.defaultProps = {
  prefixCls: 'w-prismjs',
  code: ''
};
ReactPrismjs.propTypes = {
  prefixCls: _propTypes.default.string,
  language: _propTypes.default.string,
  code: _propTypes.default.string
};
module.exports = exports.default;
